src/foam/nanos/crunch/predicate/StatusWas.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.crunch.predicate',
+  name: 'StatusWas',
+  extends: 'foam.mlang.predicate.AbstractPredicate',
+  implements: ['foam.core.Serializable'],
+
+  javaImports: [
+    'foam.core.X',
+    'foam.dao.DAO',
+    'foam.nanos.crunch.CrunchService',
+    'foam.nanos.crunch.UserCapabilityJunction',
+    'static foam.nanos.crunch.CapabilityJunctionStatus.*',
+  ],
+
+  properties: [
+    {
+      name: 'status',
+      class: 'Enum',
+      of: 'foam.nanos.crunch.CapabilityJunctionStatus'
+    }
+  ],
+
+  methods: [
+    {
+      name: 'f',
+      javaCode: `
+        if ( ! ( obj instanceof X ) ) return false;
+        var x = (X) obj;
+
+        UserCapabilityJunction oldUCJ = (UserCapabilityJunction) x.get("OLD");
```

Comment:
Assuming this happening in a rule, this entire model can be handled with FScriptPredicate
n.status==o.status

Reply:
@jlhughes does this work for enum values? My rule doesn't run after making this change

Reply:
yes. works for enums. 
to test a particular enum  then you need the package 
n.status==foam.nanos.notification.email.Status.DRAFT
for example where n of RuleData is EmailMessage.

Reply:
Realize why it doesn't work. 
n and o here come from RuleData model, 
You are passing another model, and you can reference it's properties with FScriptPredicate
so whatever your old and new properties are called. 
old.status   - for example.

Reply:
Kristina told me FScriptPredicate doesn't work at all when mixed with other predicates, I'm guessing because I can't call `x.get("OLD")`. It might have been better to have the RuleData model created within FScriptPredicate's `f` method but I'll bet there's a reason it wasn't done like that.